src/UnitTests/ExtensionMethods.cs

```diff
@@ -7,6 +7,47 @@
 
 namespace AutoMapper.UnitTests
 {
+    interface IGeneric<T> { }
+    public class When_an_extension_methods_contraints_fail : NonValidatingSpecBase
+    {
+        class Source : IGeneric<int>
+        {
+        }
+        class Destination
+        {
+            public int Count { get; set; }
+        }
+        protected override MapperConfiguration CreateConfiguration() => new(c =>
+        {
+            c.IncludeSourceExtensionMethods(typeof(GenericExtensions));
+            c.CreateMap<Source, Destination>();
+        });
+        [Fact]
+        public void It_should_fail_validation() => new Action(AssertConfigurationIsValid).ShouldThrow<AutoMapperConfigurationException>()
+            .Errors[0].UnmappedPropertyNames[0].ShouldBe(nameof(Destination.Count));
```

Comment:
Letting the exception escape would result in this test failing on `Seal` with the initial error. I wouldn't mind that :) Certainly easier to understand what's going on.

Reply:
Ha yeah, definitely